Sources/Networking/HTTPClient/HTTPResponse.swift

```diff
@@ -14,48 +14,108 @@
 
 import Foundation
 
-struct HTTPResponse<Body: HTTPResponseBody> {
+/// A type that represents an HTTP response
+protocol HTTPResponseType {
+
+    associatedtype Body: HTTPResponseBody
 
     typealias Result = Swift.Result<Self, NetworkError>
     typealias Headers = [AnyHashable: Any]
 
-    var statusCode: HTTPStatusCode
+    var statusCode: HTTPStatusCode { get }
     /// Because this property is a standard Swift dictionary, its keys are case-sensitive.
     /// To perform a case-insensitive header lookup, use the `value(forHeaderField:)` method instead.
+    var responseHeaders: HTTPClient.ResponseHeaders { get }
+    var body: Body { get }
+    var requestDate: Date? { get }
+
+}
+
+struct HTTPResponse<Body: HTTPResponseBody>: HTTPResponseType {
+
+    var statusCode: HTTPStatusCode
     var responseHeaders: HTTPClient.ResponseHeaders
     var body: Body
     var requestDate: Date?
-    var verificationResult: VerificationResult
```

Comment:
This is removed from `HTTPResponse` now, so it no longer defaults to `.notRequested`.